src/TestHeaders.am

```diff
@@ -5,21 +5,13 @@
 ## Please see the COPYING and CONTRIBUTORS files for details.
 ##
 
-## Test ./*.h files in a directory using test-suite/testheaders.sh
-
-## TODO: include in src/Common.am when all testHeaders users are the same.
-
 TESTS += testHeaders
+# append to the variable EXCLUDE_FROM_HDR_TESTING with space-separated list
+# of header files to be skipped in the testing phase.
+# NOTE: only files in the same directory as Makefiule.am are supported
 
```

Comment:
I support documenting EXCLUDE_FROM_HDR_TESTING limitations, of course!

```sh
if test "$(EXCLUDE_FROM_HDR_TESTING)" != "" && echo "$(EXCLUDE_FROM_HDR_TESTING)" | $(TR) ' ' '\n' | $(FGREP) -x -q "$$SrcFileName" ; then \
    echo "header-test: skip - $$SrcFilePath"; \
    exit 0; \
fi
```

> EXCLUDE_FROM_HDR_TESTING does, indeed, only work for header files that are in the same directory as the Makefile.am

Can you explain why, please? AFAICT based on the shell code quoted above, EXCLUDE_FROM_HDR_TESTING can exclude any file listed in SOURCES and in other testHeaders dependencies, regardless of that file path. We just need to make sure the filename (including the path) matches (from fgrep point of view).

Reply:
Because in the exclusion decision we are using the basename. So it would work if all of the following are true:
- the file name is unique in the subtree the Makefile.am belongs to
- the file name only is mentioned in the exclusion list